packages/blocks/google-cloud/src/lib/actions/get-recommendations-cli-action.ts

```diff
@@ -0,0 +1,304 @@
+  try {
+    results = await runCommands(commands, auth, useHostSession);
+  } catch (error) {
+    for (const { key } of scopeConfigs) {
+      dropdowns[key as keyof typeof dropdowns].options = {
+        disabled: true,
+        options: [],
+        placeholder: `Error fetching ${key}s`,
+        error: `${error}`,
+      };
+    }
+    return getScopeOptionsReturnStructure(dropdowns);
+  }
+
+  scopeConfigs.forEach(({ key, extractLabelAndValue }, index) => {
+    try {
+      const parsedItems = JSON.parse(results[index] ?? '[]');
+      dropdowns[key as keyof typeof dropdowns].options = {
+        disabled: false,
+        options: parsedItems.map(extractLabelAndValue),
+      };
+    } catch (error) {
+      dropdowns[key as keyof typeof dropdowns].options = {
+        disabled: true,
+        options: [],
+        placeholder: `Error parsing ${key} results`,
+        error: `${error}`,
+      };
+    }
+  });
+
+  return getScopeOptionsReturnStructure(dropdowns);
+}
+
+function getScopeOptionsReturnStructure(dropdowns: Record<string, any>) {
+  return [
+    {
+      label: 'Filter by Billing Account',
+      value: { billingAccount: dropdowns['billingAccount'] },
+    },
+    {
+      label: 'Filter by Organization ID',
+      value: { organization: dropdowns['organization'] },
+    },
+    { label: 'Filter by Project ID', value: { project: dropdowns['project'] } },
+    {
+      label: 'Filter by Folder ID',
+      value: {
+        folder: Property.ShortText({
+          displayName: 'Folder ID',
+          description:
+            'The Google Cloud Platform folder ID to use for this invocation.',
+          required: true,
+        }),
+      },
+    },
+  ];
+}
+
+function createDropdown(
+  displayName: string,
+  description: string,
+  required: boolean,
+) {
+  return Property.StaticDropdown({
+    displayName,
+    description,
+    required,
+    options: {
+      options: [],
+    },
+  });
+}
+
+function getRecommendersDropdown() {
+  return Property.MultiSelectDropdown({
+    displayName: 'Recommender',
+    description:
+      'The Google Cloud Platform recommenders to use for this invocation.',
+    required: true,
+    refreshers: [
+      'auth',
+      'useHostSession',
+      'useHostSession.useHostSessionCheckbox',
+      'filterByProperty',
+      'filterByProperty.project',
+      'filterByProperty.organization',
+      'filterByProperty.billingAccount',
+    ],
+    options: async ({ auth, useHostSession, filterByProperty }) => {
+      const shouldUseHostCredentials =
+        (useHostSession as { useHostSessionCheckbox?: boolean })
+          ?.useHostSessionCheckbox === true;
+
+      if (!auth && !shouldUseHostCredentials) {
+        return {
+          disabled: true,
+          options: [],
+          placeholder: 'Please authenticate to see recommenders.',
+          error: undefined,
+        };
+      }
+      
+      let filterFlag = '';
+      if (!filterByProperty['billingAccount']) {
```

Comment:
So, recommenders list does not support billing account as a param, it does allow folder, org and project though